backend/server/compute/diffexp_generic.py

```diff
@@ -65,16 +67,61 @@ def diffexp_ttest_from_mean_var(meanA, varA, nA, meanB, varB, nB, top_n, diffexp
 
     # logfoldchanges: log2(meanA / meanB)
     logfoldchanges = np.log2(np.abs((meanA + 1e-9) / (meanB + 1e-9)))
+    stats_to_sort = np.abs(tscores)
 
     # find all with lfc > cutoff
-    lfc_above_cutoff_idx = np.nonzero(np.abs(logfoldchanges) > diffexp_lfc_cutoff)[0]
-    stats_to_sort = np.abs(tscores)
 
-    # derive sort order
-    if lfc_above_cutoff_idx.shape[0] > top_n:
+    if two_lists:
```

Comment:
This code seems problematic:
* partitions twice, which is inefficient
* lots of redundancy

Why not just replace it with something which leverages a single (multi-pivot) partition, sorting by natural order (not abs), and just cuts from top & bottom of sorted list?

(caveat: typed this in without testing, so treat as psuedo-code)
(caveat: this is only the two_list variant, and it inlines derive_sort_order())

```
stats_to_sort = tscores
lfc_above_cutoff_idx = np.nonzero(np.abs(logfoldchanges) > diffexp_lfc_cutoff)[0]
rel_t_partition = np.argpartition(stats_to_sort[lfc], (top_n, -top_n))[-top_n:][:top_n]
t_partition = lfc[rel_t_partition]
rel_sort_order = np.argsort(stats_to_sort[t_partition])[::-1]
sort_order = t_partition[rel_sort_order]

logfoldchanges_top_n = logfoldchanges[sort_order]
pvals_top_n = pvals[sort_order]
pvals_adj_top_n = pvals_adj[sort_order]

result = {
    "positive": [[sort_order[i], logfoldchanges_top_n[i], pvals_top_n[i], pvals_adj_top_n[i]] for i in range(top_n)],
    "negative": [
        [sort_order[i], logfoldchanges_top_n[i], pvals_top_n[i], pvals_adj_top_n[i]] for i in range(-1, -1 - top_n, -1)
    ],
}
```